### PR TITLE
fix: support uppercase when parsing `Interval`

### LINKED ISSUE #8465: select interval '1 YEAR' error
### Describe the bug

```
> select interval '1 YEAR' 
Arrow error: Not yet implemented: Unsupported Interval Expression with value "1 YEAR seconds"
```
however use lowercase is ok.
```
> select interval '1 year'
+--------------------------------------------------------+
| IntervalMonthDayNano("950737950171172051122527404032") |
+--------------------------------------------------------+
| 0 years 12 mons 0 days 0 hours 0 mins 0.000000000 secs |
+--------------------------------------------------------+
```
I try other databases such as duckdb and postgres, and uppercase style is ok.

### To Reproduce

use datafusion-cli and `select interval '1 YEAR'`
DataFusion CLI v33.0.0

### Expected behavior

ouput the right result.

### Additional context

The reason may be because https://github.com/apache/arrow-datafusion/blob/main/datafusion/sql/src/expr/value.rs#L319
the `has_units` function just check the lowercase style, and I think add `to_lowercase` in it can solve this problem.

### PR DESCRIPTION
**Pull request by QuenKar**
## Which issue does this PR close?

Closes #8465 .

## What changes are included in this PR?

convert the string value in `has_units` to lowercase.

## Are these changes tested?

yes

## Are there any user-facing changes?

yes

### CODE REVIEW

File: datafusion/sql/src/expr/value.rs [hunk 1]
```diff
@@ -468,4 +469,38 @@ mod tests {
...
+            "1 microsecond",
+            "1 microseconds",
+            "1 nanosecond",
+            "1 nanoseconds",
+        ];
+        for case in cases {
+            assert!(has_units(case));
+            assert!(has_units(case.to_uppercase().as_str()));
+        }
+    }
```
Comment (Weijun-H) [inline datafusion/sql/src/expr/value.rs:505]: Are these tests necessary?

Comment (QuenKar) [reply datafusion/sql/src/expr/value.rs:505]: it's a little redundant, I will delete them.

**Review by waynexia — APPROVED** [review]
Looking good 👍 Thanks @QuenKar

**Review by alamb — APPROVED** [review]
Thank you @QuenKar  and @waynexia

**Pull request merged**